rstudio/rstudio
# allow user to modify rendering engine in startup window

### Intent

Addresses part of https://github.com/rstudio/rstudio/issues/10695.
Also part of https://github.com/rstudio/rstudio/issues/10929, for `RS_LOG_LEVEL`.

### Approach

Ports the rendering engine portion of the "Choose R" dialog, and uses that to set the rendering engine option. The plumbing is mostly straightforward; the main difference is around what data gets marshaled around in the Choose R dialog (now, we pass along the selected rendering engine, and that gets handled after the user requests a version of R).

### Automated Tests

Not included; it isn't trivial to test restart of the application.

### QA Notes

To test, from a console, trying running:

```
set RSTUDIO_DESKTOP_PROMPT_FOR_R=TRUE
C:\path\to\rstudio.exe
```

And confirm that you see a select box for changing the rendering engine.

<img width="415" alt="Screen Shot 2022-04-07 at 1 01 23 PM" src="https://user-images.githubusercontent.com/1976582/162286430-8e8f424d-6cdd-446d-ae69-4d4c9efeedc0.png">

And check that Help -> Diagnostics -> Show GPU Diagnostics matches the selection. For example, for Software rendering:

<img width="807" alt="Screen Shot 2022-04-07 at 1 07 10 PM" src="https://user-images.githubusercontent.com/1976582/162287301-e24698dc-400b-479d-bee9-ae90056287c9.png">


### Checklist
